src/tempo.py

```diff
@@ -83,91 +63,29 @@ def tempo_grpc_server_port(self) -> int:
 
-    def get_external_ports(self, service_name_prefix: str) -> List[Tuple[str, int, int]]:
-        """List of service names and port mappings for the kubernetes service patch.
-
-        Includes the tempo server as well as the receiver ports.
-        """
-        # todo allow remapping ports?
-        all_ports = {**self.server_ports}
-        return [
-            (
-                (f"{service_name_prefix}-{service_name}").replace("_", "-"),
-                all_ports[service_name],
-                all_ports[service_name],
-            )
-            for service_name in all_ports
-        ]
-
-    @property
-    def url(self) -> str:
-        """Base url at which the tempo server is locally reachable over http."""
-        scheme = "https" if self.use_tls else "http"
-        return f"{scheme}://{self._external_hostname}"
-
-    def get_receiver_url(self, protocol: ReceiverProtocol, ingress: TraefikRouteRequirer):
-        """Return the receiver endpoint URL based on the protocol.
-
-        if ingress is used, return endpoint provided by the ingress instead.
-        """
-        protocol_type = receiver_protocol_to_transport_protocol.get(protocol)
-        # ingress.is_ready returns True even when traefik hasn't sent any data yet
-        has_ingress = ingress.is_ready() and ingress.external_host and ingress.scheme
-        receiver_port = self.receiver_ports[protocol]
-
-        if has_ingress:
-            url = (
-                ingress.external_host
-                if protocol_type == "grpc"
-                else f"{ingress.scheme}://{ingress.external_host}"
-            )
-        else:
-            url = self._external_hostname if protocol_type == "grpc" else self.url
-
-        return f"{url}:{receiver_port}"
-
-    def _build_server_config(self):
-        server_config = tempo_config.Server(
-            http_listen_port=self.tempo_http_server_port,
-            # we need to specify a grpc server port even if we're not using the grpc server,
-            # otherwise it will default to 9595 and make promtail bork
-            grpc_listen_port=self.tempo_grpc_server_port,
-        )
-
-        if self.use_tls:
-            server_tls_config = tempo_config.TLS(
-                cert_file=str(self.tls_cert_path),
-                key_file=str(self.tls_key_path),
-                client_ca_file=str(self.tls_ca_path),
-            )
-            server_config.http_tls_config = server_tls_config
-            server_config.grpc_tls_config = server_tls_config
-
-        return server_config
-
-    def generate_config(
+    def config(
         self,
-        receivers: Sequence[ReceiverProtocol],
-        s3_config: dict,
-        roles_addresses: Dict[str, Set[str]],
-        peers: Optional[Set[str]] = None,
-    ) -> Dict[str, Any]:
+        coordinator: Coordinator,
+    ) -> str:
         """Generate the Tempo configuration.
 
         Only activate the provided receivers.
         """
+
         config = tempo_config.Tempo(
```

Comment:
wait, do we have a `tempo.Tempo` and a `tempo_config.Tempo`? 
Confusing.

Reply:
Renamed to TempoConfig